### PR TITLE
Fixes the brushing issue reported at #26

### PR DESCRIPTION
**Pull request by victorfern91**

### CODE REVIEW

File: src/DensityChart/DensityChart.js [hunk 1]
```diff
@@ -73,11 +74,24 @@ export default class DensityChart extends PureComponent {
     }
 
     componentDidUpdate(prevProps) {
-        const { brushDomainMin, brushDomainMax, densityChartXScale } = this.props;
+        let min = this.props.brushDomainMin;
+        let max = this.props.brushDomainMax;
+
+        const { densityChartXScale } = this.props;
+
+        if (max >= this.props.timeDomainMax) {
```
Comment (lfac-pt) [inline src/DensityChart/DensityChart.js]: What is this for?

File: src/Histogram/Histogram.js [hunk 2]
```diff
@@ -480,17 +487,17 @@ export class Histogram extends PureComponent {
...
-                width={densityChartDimensions.width}
-                height={densityChartDimensions.height}
+                width={this.state.densityChartDimensions.width}
+                height={this.state.densityChartDimensions.height}
                 padding={PADDING}
-                brushDomainMax={dateToTimestamp(brushDomain.max)}
-                brushDomainMin={dateToTimestamp(brushDomain.min)}
+                brushDomainMin={this.state.brushDomain.min}
+                brushDomainMax={this.state.brushDomain.max}
+                timeDomainMax={this.state.timeDomain.max}
```
Comment (lfac-pt) [inline src/Histogram/Histogram.js]: What is the difference between timeDomain and brushDomain?

Comment (victorfern91) [reply src/Histogram/Histogram.js]: Ok, I need to add more information to this! 

So, the `timeDomain` is related to the full range of the data  given to the histogram component.

Please see this example
```json
[
    { "timestamp":  1, "total": 1  },
    { "timestamp":  2, "total": 1  },
    { "timestamp":  3, "total": 1  },
    { "timestamp":  4, "total": 1  },
    { "timestamp":  5, "total": 1  },
    { "timestamp":  6, "total": 1  }
]
```

The `timeRange` has  the domain between `[ 1, 6 ]`  and the `brushDomain` could be anything between this values  .

File: src/DensityChart/DensityChart.js (hunk 1 above)
Comment (victorfern91) [reply src/DensityChart/DensityChart.js]: This is used in order to  not be possible drag a brush area to values  outside of the `timeDomain` imposed by the data

Comment (lfac-pt) [reply src/DensityChart/DensityChart.js]: But how could that happen? The maximum value of the density chart is the maximum of the data

File: src/Histogram/Histogram.js (hunk 2 above)
Comment (lfac-pt) [reply src/Histogram/Histogram.js]: I think the name `timeDomain` is not very good, the `brushDomain` is also a time domain. Why not `overallDomain` or `overallDataDomain` or something that indicates that this related to all the data?

File: src/DensityChart/DensityChart.js (hunk 1 above)
Comment (victorfern91) [reply src/DensityChart/DensityChart.js]: did you have seen the gif at #26 ?

File: stories/interactive.stories.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/**
```
Comment (lfac-pt) [inline stories/interactive.stories.js]: We aren't putting headers in nowhere else, why put one here?

File: stories/interactive.stories.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * @author Victor Fernandes (victor.fernandes@feedzai.com)
+ * @since 1.7.0
+ */
+
+const stories = storiesOf("Histogram - Interactive data", module);
+
+stories.addDecorator(withKnobs);
+
+stories
+    .add("Basic example", () => (
```
Comment (lfac-pt) [inline stories/interactive.stories.js]: Why didn't you add this to the other stories file?

**Comment by victorfern91** [comment]
> @victorfern91 do not forget to update the compiled storybook and the changelog

Done 😎

File: stories/interactive.stories.js (hunk 4 above)
Comment (lfac-pt) [reply stories/interactive.stories.js]: The others stories are also interactive, I don't see a difference

Comment (lfac-pt) [reply stories/interactive.stories.js]: I don't why this couldn't be the base example in the other storybook file

**Pull request merged**